### PR TITLE
CClient - TLS socket and https client. 

### LINKED ISSUE #724: cclient: support HTTPS connections
Many IRI nodes are only exposed via HTTPS nowadays.
We should strive to support this as well.

### PR DESCRIPTION
**Pull request by oopsmonk**
This PR implemented tls socket and https client, fixed #724 
Removed `utils/containers/lock_free`

# Test Plan:
CI pass

### CODE REVIEW

File: cclient/http/http.h [hunk 1]
```diff
@@ -15,6 +15,15 @@ extern "C" {
 #include <stdlib.h>
 #include "cclient/service.h"
 
+// socket buffer can overwrite thought config
```
Comment (thibault-martinez) [inline cclient/http/http.h]: through ?

File: utils/handles/socket.c [hunk 2]
```diff
@@ -44,3 +46,135 @@ int open_client_socket(char const *const hostname, const size_t port) {
...
+}
+
+int tls_socket_connect(mbedtls_ctx_t *tls_ctx, char const *host, uint16_t port,
+                       char const *ca_pem, char const *client_cert_pem,
+                       char const *client_pk_pem, retcode_t *error) {
+  int mbedtls_ret = -1;
+  char const drgb_pres[] = "iota_tls_client";
+  bool is_client_auth = false;
+  char port_string[6];
+  sprintf(port_string, "%d", port);
```
Comment (thibault-martinez) [inline utils/handles/socket.c:90]: As `port_string` is not guaranteed to be 0-initialized, shouldn't we do something like `sprintf(port_string, "%06d", port);` ?

File: cclient/http/http.c [hunk 3]
```diff
@@ -165,25 +229,45 @@ retcode_t iota_service_query(const void* const service_opaque,
...
-      RC_OK) {
-    goto cleanup;
-  }
-  if ((result = read_data_from_iota_service(sockfd, response)) != RC_OK) {
-    goto cleanup;
-  }
-cleanup:
-  if (sockfd != -1) {
-    close_socket(sockfd);
+    if ((result = send_headers_to_iota_service(sockfd, http_settings,
```
Comment (thibault-martinez) [inline cclient/http/http.c]: Can we uniform names between http and https ?
- `connect_to_iota_service ` and `tls_socket_connect `
- `close_socket `  and  `tls_socket_close `
- `send_headers_to_iota_service ` and `https_request_header `
- `send_data_to_iota_service ` and `https_request_data `
- `read_data_from_iota_service `and `https_response_read `

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: cclient/http/http.h (hunk 1 above)
Comment (oopsmonk) [reply cclient/http/http.h]: a gcc preprocessor macro

Comment (thibault-martinez) [reply cclient/http/http.h]: I meant, `thought` is a typo ?

Comment (oopsmonk) [reply cclient/http/http.h]: haha..typo

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**